lightning_client.go

```diff
@@ -3757,3 +3762,62 @@ func (s *lightningClient) SendCustomMessage(ctx context.Context,
 	_, err := s.client.SendCustomMessage(rpcCtx, rpcReq)
 	return err
 }
+
+// SubscribeCustomMessages subscribes to a stream of custom messages, optionally
+// filtering by peer and message type. The channels returned will be closed
+// when the subscription exits.
+func (s *lightningClient) SubscribeCustomMessages(ctx context.Context) (
+	<-chan CustomMessage, <-chan error, error) {
+
+	rpcCtx := s.adminMac.WithMacaroonAuth(ctx)
+	rpcReq := &lnrpc.SubscribeCustomMessagesRequest{}
+
+	client, err := s.client.SubscribeCustomMessages(rpcCtx, rpcReq)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var (
+		// Buffer error channel by 1 so that consumer reading from this
+		// channel does not block our exit.
+		errChan = make(chan error, 1)
+		msgChan = make(chan CustomMessage)
+	)
+
+	s.wg.Add(1)
+	go func() {
+		defer func() {
+			s.wg.Done()
+			close(errChan)
```

Comment:
nit: maybe we could change the order, close channels first then waitgroup.

Reply:
[Changed](https://github.com/lightninglabs/lndclient/compare/ebeabd076237480658be58ec122132809eb9d53a..a792c60eb81aa94d804ace90bff3e95eb9b5ceb9) 👍